enterprise/fulltext-addon/src/main/java/org/neo4j/kernel/api/impl/fulltext/lucene/ReadOnlyFulltext.java

```diff
@@ -17,27 +17,17 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program. If not, see <http://www.gnu.org/licenses/>.
  */
-package org.neo4j.kernel.api.impl.fulltext;
+package org.neo4j.kernel.api.impl.fulltext.lucene;
 
-/**
- * Fulltext index type.
- */
-public enum FulltextIndexType
+public interface ReadOnlyFulltext extends AutoCloseable
 {
-    NODES
-            {
-                @Override
-                public String toString()
-                {
-                    return "Nodes";
-                }
-            },
-    RELATIONSHIPS
-            {
-                @Override
-                public String toString()
-                {
-                    return "Relationships";
-                }
-            }
+    /**
+     * Queires the fulltext index with the given lucene-syntax query
+     * @param query the lucene query
+     * @return A {@link ScoreEntityIterator} over the results
+     */
+    ScoreEntityIterator query( String query );
+
+    @Override
+    void close();
```

Comment:
this interface overrides AutoCloseable close to pretend it does not throw anything while two implementations do that in reality and try to catch and rethrow exceptions while they do not need to do that since in a places where its used already we are free to throw IOExceptions since lucene do that anyway